### PR TITLE
Custom artifact name

### PR DESCRIPTION
**Pull request by pilshchikov**

### CODE REVIEW

File: src/tiden/artifacts.py [hunk 1]
```diff
@@ -390,6 +390,12 @@ def repack(artifact_name, src_path, checksum, work_dir, rules, artifacts_dir, ar
                 mkdir(extract_dir)
                 with data["open"](src_path, data["mode"]) as arch:
                     arch.extractall(extract_dir)
+
+                # custom artifact name
+                name = artifacts[artifact_name].get('name')
+                if name:
+                    src_path = f'{name}{data["pattern"]}'
```
Comment (iartiukhov) [inline src/tiden/artifacts.py:397]: I did not get the idea. `src_path` is an absolute path to an ZIP artifact and I'd suggest not to touch it. Shouldn't we change `new_zip_file` instead?

Comment (pilshchikov) [reply src/tiden/artifacts.py:397]: `src_path` here it's a base name for `new_zip_file`
If we are trying to change `new_zip_file` on 400 line then we need more code to detect `.replace.zip` and don't touch that part
Change `scr_path` it's the easy way to do it

Comment (iartiukhov) [reply src/tiden/artifacts.py:397]: Easier but dirtier. This code has less WTFs IMO:
```
custom_name = artifacts[artifact_name].get('name')
new_zip_file_name = f'{custom_name}{data["pattern"]}' if custom_name else basename(src_path)
new_zip_file_name = new_zip_file_name.replace(data["pattern"], f'.repack.data["pattern"]')
new_zip_file = join(work_dir, new_zip_file_name)
```

Comment (pilshchikov) [reply src/tiden/artifacts.py:397]: ok, it looks pretty now

**Review by iartiukhov — APPROVED** [review]
LGTM!

**Pull request merged**